### PR TITLE
Add FileExists assertion

### PR DESCRIPTION
**Pull request by ghost**
Like proposed at https://github.com/stretchr/testify/issues/428

### CODE REVIEW

File: require/require.go [hunk 1]
```diff
@@ -258,6 +258,28 @@ func Falsef(t TestingT, value bool, msg string, args ...interface{}) {
 	}
 }
 
+// FileExists asserts that the filename passed represents an existing file.
+//
+//  assert.FileExists(t, "main.go")
```
Comment (hanzei) [inline require/require.go]: This should be `require.FileExists(t, "main.go")`

File: assert/assertions.go [hunk 2]
```diff
@@ -1120,6 +1121,24 @@ func NotZero(t TestingT, i interface{}, msgAndArgs ...interface{}) bool {
...
+//
+// Returns whether the assertion was successful (true) or calls Fail.
+func FileExists(t TestingT, filename string, msgAndArgs ...interface{}) bool {
+	_, err := os.Lstat(filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return Fail(t, fmt.Sprintf("unable to find file %q (%s)", filename, err), msgAndArgs...)
+		}
+
+		return Fail(t, fmt.Sprintf("error when running os.Lstat(%q) (%s)", filename, err), msgAndArgs...)
```
Comment (hanzei) [inline assert/assertions.go]: This error message seams a bit to technical. You might want to change it to `fmt.Sprintf("error when reading file %q: %s", filename, err), msgAndArgs...)`

File: assert/assertions.go [hunk 3]
```diff
@@ -1120,6 +1121,24 @@ func NotZero(t TestingT, i interface{}, msgAndArgs ...interface{}) bool {
...
+// FileExists asserts that the filename passed represents an existing file.
+//
+//  assert.FileExists(t, "main.go")
+//
+// Returns whether the assertion was successful (true) or calls Fail.
+func FileExists(t TestingT, filename string, msgAndArgs ...interface{}) bool {
+	_, err := os.Lstat(filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return Fail(t, fmt.Sprintf("unable to find file %q (%s)", filename, err), msgAndArgs...)
```
Comment (hanzei) [inline assert/assertions.go]: Why should `err` be printed here?

**Review by hanzei — COMMENTED** [review]
Good addition. Just a few comments.

**Comment by hanzei** [comment]
Requesting a 2nd review from @dignifiedquire

**Comment by hanzei** [comment]
Could you please rebase onto master and see if this fixes the CI issues?

File: assert/assertions.go (hunk 3 above)
Comment (ghost) [reply assert/assertions.go]: I took a look and I thought to return the error 'cause IMHO the testing is easier.
What do you think @Hanzei?

**Comment by ghost** [comment]
I tried to run Travis CI on my fork and it passes: https://travis-ci.org/domcorvasce/testify.

**Pull request closed without merging**